test/api/detection_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <fstream>
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/io/exception.hpp>
+#include <seqan3/io/sam_file/sam_flag.hpp>
+
+#include "iGenVar.hpp"
+
+#include "modules/sv_detection_methods/analyze_cigar_method.hpp"    // for the split read method
+#include "modules/sv_detection_methods/analyze_sa_tag_method.hpp"   // for the cigar string method
+#include "structures/junction.hpp"                                  // for class Junction
+
+using seqan3::operator""_cigar_operation;
+using seqan3::operator""_dna5;
+
+/* -------- detection methods tests -------- */
+
+// TODO (irallia): implement test cases
+
+TEST(junction_detection, cigar_string_method_simple)
+{
+    std::string const read_name = "read021";
+    std::string const chromosome = "chr1";
+    int32_t const query_start_pos = 1;
+    std::vector<seqan3::cigar> cigar_sting = {{5, 'S'_cigar_operation}, {15, 'M'_cigar_operation},
+                                              {6, 'D'_cigar_operation}, {24, 'M'_cigar_operation}}; //5S15M6D24M
+    seqan3::dna5_vector seq = {"GGGCTCATCGATCGATTTCGGATCGGGGGGCCCCCATTTTAAACGGCCCC"_dna5};
```

Comment:
The CIGAR string indicates a sequence size of 5+15+24=44 but the sequence itself is 50bp long. The simplest solution would be to extend the final match operation from 24 to 30.